apm-agent-plugins/apm-process-plugin/src/test/java/co/elastic/apm/agent/process/ProcessHelperTest.java

```diff
@@ -57,7 +58,7 @@ void before() {
         TransactionUtils.fillTransaction(transaction);
 
         storageMap = WeakConcurrent.buildMap();
-        helper = new ProcessHelper(storageMap);
+        helper = new ProcessHelper(new WeakMapReferenceCounter<>(storageMap));
```

Comment:
This won't work anymore if you hide `WeakMapReferenceCounter` as I suggested.
I would suggest making `ReferenceCounter` extend `Iterable<Map.Entry<K,V>>` and to just use `tracer.createReferenceCounter()` in this test.

Reply:
That might break the test as the references are weak but the keys are not referenced in the test. So the test would turn flakey. But it can be trivially solved by using an anonymous class that delegates to the non-weak-map.